### PR TITLE
Abstract task

### PR DESCRIPTION
**Pull request by marc-hanheide**
This is adding a new class `AbstractTaskServer` that other actionlib servers that implement schedulable tasks _should_ use. The rationale is that the abstract class also offers an `<actionname>_create` service that routines can call to create a suitable `Task` object. So it offer a factory service. The idea here is that for instance the `gcal_routine` can then call this factory to automatically create a suitable task object just from a given name. The task objects created by the factory service should have all fields populated with useful defaults, so that in most cases on can simply send this to the `task_executor`. 
I have refactored `wait_action` to actually use the new base class. Another nice side effect of this is that it also make sure that the correct `<actionname>_interuptible` service is made available. 

Please @hawesie let me know what you think and if this is acceptable.

### CODE REVIEW

File: wait_action/scripts/wait_node.py [hunk 1]
```diff
@@ -29,6 +19,12 @@ def end_wait(self, req):
             self.server.set_preempted()
         return EmptyResponse()
 
+    def create(self, req):
```
Comment (marc-hanheide) [inline wait_action/scripts/wait_node.py:22]: this is the factory method specialised for the subclass.

File: wait_action/scripts/wait_node.py [hunk 2]
```diff
@@ -29,6 +19,12 @@ def end_wait(self, req):
...
+        t = super(WaitServer, self).create(req)
+        task_utils.add_time_argument(t, rospy.Time())
+        task_utils.add_duration_argument(t, t.max_duration)
+        return t
+        
     def execute(self, goal):
```
Comment (marc-hanheide) [inline wait_action/scripts/wait_node.py:28]: this is specialised for the subclass

File: wait_action/scripts/wait_node.py [hunk 3]
```diff
@@ -49,7 +45,7 @@ def execute(self, goal):
         rospy.loginfo("target wait time: %s" % datetime.fromtimestamp(target.secs))	
 
         # how often to provide feedback 
-        feedback_secs = 5
+        feedback_secs = 5.0
```
Comment (marc-hanheide) [inline wait_action/scripts/wait_node.py:48]: there was a div by zero for short durations. This fixes it.

File: strands_executive_msgs/srv/CreateTask.srv [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# This service returns the currently executing task
```
Comment (hawesie) [inline strands_executive_msgs/srv/CreateTask.srv:1]: No it doesn't.

**Comment by hawesie** [comment]
I don't really see the benefit of the factory service in the general case. The fiddly bit of task creation is assigning the parameters correctly from local data. The service, by its nature, cannot do this, except if you allow the action to be used to dictate the parameters, which is not something I think will be widely used. If that is the behaviour you want then I suggest a service which just takes a string as input and it returns the fully instantiated task.

**Comment by marc-hanheide** [comment]
right! That makes sense. I will revise this so that it takes a YAML string as input.

**Comment by marc-hanheide** [comment]
So, @hawesie This now uses a YAML argument in the factory service. That's indeed more useful. I still need to improve the way the extra arguments are handled and supported. It does work, but probably not generic enough.
The YAML argument can be left empty, then a default task is created. If it's not empty then the values that are defined in YAML overwrite any defaults. So not the whole task needs to be specified, only individual slots need to be defined.

**Comment by marc-hanheide** [comment]
I would like to merge this now in order to base our AAF actions on it. We might still see an improvement of the factory service, but I want this to be available so people can implement the tasks. If that's ok @hawesie ?

**Comment by hawesie** [comment]
Ok, but the final comment I will make is that this should not be viewed as a required superclass for task providers, only an optional one. IMO one of the strengths of the task system is that it doesn't require providers to implement anything beyond the usual actionlib interfaces.

**Comment by marc-hanheide** [comment]
I see your point entirely, the problem is that I have seen these functionalities being re-implemented at various places. I'm not a strong advocate of class inheritance myself, but I believe this to be useful. If you have a better idea I'm happy to learn it.

**Comment by hawesie** [comment]
I'm entirely happy that this functionality is available for those that want it, and I don't think there is a better way to provide support for this kind of thing, but I just want to make sure we don't end up putting something in here at a later date that becomes a requirement.

**Pull request merged**